src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -61,6 +61,14 @@ public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageConte
         }
 
         /// <summary>
+        /// Creates a <see cref="IIncomingPhysicalMessageContext"/> based on the current context.
+        /// </summary>
+        public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageContext(this StageConnector<ITransportReceiveContext, IIncomingPhysicalMessageContext> stageConnector, IncomingMessage incomingMessage, ITransportReceiveContext sourceContext)
```

Comment:
Sorry that my feedback comes in intervals. I will make sure this will improve in the future. But I think this is no longer used but instead we should have an extension method which allows to create an `ISatelliteProcessingContext` which should be used https://github.com/Particular/NServiceBus/pull/3516/files#diff-75593bed7912b67139c67010b2d65a15R11. If we don't have this there is no way users could replace this behavior.